src/types.rs

```diff
@@ -159,12 +163,58 @@ impl fmt::Display for Problem {
             write!(f, " ({})", r#type)?;
         }
 
+        if let Some(subproblems) = &self.subproblems {
+            let count = subproblems.len();
+            write!(f, ": {count} subproblems: ")?;
+            for (i, subproblem) in subproblems.iter().enumerate() {
+                write!(f, "{subproblem}")?;
+                if i != count - 1 {
+                    f.write_str(", ")?;
+                }
+            }
+        }
+
         Ok(())
     }
 }
 
 impl std::error::Error for Problem {}
 
+/// An RFC 8555 subproblem document contained within a problem returned by the ACME server
+///
+/// See <https://www.rfc-editor.org/rfc/rfc8555#section-6.7.1>
+#[derive(Clone, Debug, Deserialize)]
+pub struct Subproblem {
+    /// One of an enumerated list of problem types
+    ///
+    /// See <https://datatracker.ietf.org/doc/html/rfc8555#section-6.7>
+    pub r#type: Option<String>,
+    /// A human-readable explanation of the problem
+    pub detail: Option<String>,
+    /// The identifier associated with this problem
+    pub identifier: Option<Identifier>,
+}
+
+impl fmt::Display for Subproblem {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("error ")?;
+
+        if let Some(identifier) = &self.identifier {
+            write!(f, "for {identifier}")?;
+        }
```

Comment:
How do you feel about dropping `error for `? I think I'd prefer having less "boilerplate" inserted by this impl in favor of emphasizing the content from the server.

Reply:
SGTM.